magmi/inc/magmi_engine.php

```diff
@@ -508,7 +508,28 @@ public function connectToMagento()
                 $user = $default_setup->username;
                 $pass = $default_setup->password;
                 $port = $default_setup->port;
-            } else {
+            }else if($conn == 'envphp'){ //berto - scraping env.php
+				$baseDir = $this->getProp('MAGENTO', 'basedir');
+                $envPath = $baseDir.'/app/etc/env.php';
+                if (!file_exists($envPath)) {
+                    throw new Exception("Cannot load xml from path '$envPath'");
+                }
+				$content = rtrim(file_get_contents($envPath));
+				$db_info = array();
+				preg_match("/'host'\s*=>\s*'\w*'/",$content,$db_info['host'] );
```

Comment:
Env.php is a php file, so cant it be just included?
Using regex for this doesnt feel right.
Also please format your code (PSR-2)